types/n3/index.d.ts

```diff
@@ -21,9 +21,9 @@ export interface Prefixes<I = RDF.NamedNode> {
 export type Term = NamedNode | BlankNode | Literal | Variable | DefaultGraph;
 export type PrefixedToIri = (suffix: string) => NamedNode;
 
-export class NamedNode implements RDF.NamedNode {
+export class NamedNode<Iri extends string = string> implements RDF.NamedNode<Iri> {
     readonly termType: "NamedNode";
-    readonly value: string;
+    readonly value: Iri;
```

Comment:
I was just looking at that. Should `constructor` and `id` also change to use `Iri` type?

Reply:
Indeed, good catch.